neumai-tools/neumai_tools/PipelineCollection/PipelineCollection.py

```diff
@@ -53,4 +54,29 @@ def search_separate(self, query:str, number_of_results:int)-> List:
     def search_routed(self, query:str, number_of_results:int)-> List:
         """Routed search based on the contents available in a pipeline"""
         # Need to add descriptions to the pipeline and generate a basic index on top of them
-        raise NotImplementedError("In the works. Contact founders@tryneum.com for information")
\ No newline at end of file
+
+        pipe_to_similarity = {}
+        for pipe in self.pipelines:
+            pipe_representative = pipe.sink.get_representative_vector()
```

Comment:
This seems good. The representative vector in the future could be extended to also use the description as the representative vector in cases where there is not possible to compute one of the sink or where you would want to use that instead.